e2e-tests/tfm/cypress/e2e/journeys/case-amendments/facility-end-date/feature-flag-enabled/gef-deal-amend-facility-end-date-and-then-change-to-bank-review-date.spec.js

```diff
@@ -0,0 +1,101 @@
+import relative from '../../../../relativeURL';
+import facilityPage from '../../../../pages/facilityPage';
+import { ADMIN, BANK1_MAKER1, PIM_USER_1, T1_USER_1 } from '../../../../../../../e2e-fixtures';
+import { MOCK_FACILITY_ONE } from '../../../../../fixtures/mock-gef-facilities';
+import { MOCK_APPLICATION_AIN } from '../../../../../fixtures/mock-gef-deals';
+import { DEAL_TYPE } from '../../../../../../../gef/cypress/fixtures/constants';
+import amendmentsPage from '../../../../pages/amendments/amendmentsPage';
+import dateConstants from '../../../../../../../e2e-fixtures/dateConstants';
+
+if (Cypress.env('FF_TFM_FACILITY_END_DATE_ENABLED') === 'true') {
+  context('Amendments - GEF deal add bank review date', () => {
+    let dealId;
+    let dealFacilities;
+
+    const MOCK_GEF_FACILITY = {
+      ...MOCK_FACILITY_ONE,
+      hasBeenIssued: true,
+      isUsingFacilityEndDate: true,
+      facilityEndDate: new Date('2023-01-01'),
+      bankReviewDate: undefined,
+    };
+
+    before(() => {
+      // inserts a gef deal
+      cy.insertOneGefDeal(MOCK_APPLICATION_AIN, BANK1_MAKER1).then((insertedDeal) => {
+        dealId = insertedDeal._id;
+        // updates a gef deal so has relevant fields
+        cy.updateGefDeal(dealId, MOCK_APPLICATION_AIN, BANK1_MAKER1);
+
+        cy.createGefFacilities(dealId, [MOCK_GEF_FACILITY], BANK1_MAKER1).then((createdFacilities) => {
+          dealFacilities = createdFacilities.details;
+        });
+
+        cy.submitDeal(dealId, DEAL_TYPE.GEF, T1_USER_1);
+      });
+    });
+
+    beforeEach(() => {
+      cy.login(PIM_USER_1);
+      cy.visit(relative(`/case/${dealId}/facility/${dealFacilities._id}`));
+    });
+
+    after(() => {
+      cy.deleteDeals(dealId, ADMIN);
+    });
+
+    it('should display the current facility end date details the Details tab', () => {
+      facilityPage.facilityIsUsingFacilityEndDate().should('have.text', 'Yes');
+      facilityPage.facilityFacilityEndDate().should('have.text', '1 January 2023');
+      facilityPage.facilityBankReviewDate().should('not.exist');
+    });
+
+    it('should amend the facility correctly when first adding a facility end date and then switching to a bank review date during the amendment', () => {
+      amendmentsPage.navigateToIsUsingFacilityEndDatePage(true, false);
```

Comment:
can this it be split up, its very long

Reply:
I think it's fine - the name makes it clear what it's doing, and by splitting it up it would mean the test would have to start from scratch each time making the test run a lot longer. The only key value to check is at the end of the test so there's no logical way to split it up.

Reply:
Can you please create a tech debt ticket for refactoring this test and the others I have commented on